src/apps/KintoAppRegistry.sol

```diff
@@ -0,0 +1,271 @@
+    uint256 public constant override GAS_LIMIT_PERIOD = 30 days;
+    uint256 public constant override GAS_LIMIT_THRESHOLD = 1e16; // 0.01 ETH
+
+    /* ============ State Variables ============ */
+
+    uint256 public override appCount;
+    mapping(address => IKintoAppRegistry.Metadata) private _appMetadata;
+    mapping(address => mapping(address => bool)) private _appSponsoredContracts; // other contracts to be sponsored
+
+    mapping(address => address) public override childToParentContract;
+
+    /* ============ Events ============ */
+    event AppCreated(address indexed _app, address _owner, uint256 _timestamp);
+    event AppUpdated(address indexed _app, address _owner, uint256 _timestamp);
+    event AppDSAEnabled(address indexed _app, uint256 _timestamp);
+
+    /* ============ Constructor & Initializers ============ */
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize() external initializer {
+        __ERC721_init("Kinto APP", "KINTOAPP");
+        __ERC721Enumerable_init();
+        __Ownable_init();
+        __UUPSUpgradeable_init();
+        _transferOwnership(msg.sender);
+    }
+
+    /**
+     * @dev Authorize the upgrade. Only by the upgrader role.
+     * @param newImplementation address of the new implementation
+     */
+    // This function is called by the proxy contract when the implementation is upgraded
+    function _authorizeUpgrade(address newImplementation) internal override onlyOwner {}
+
+    /* ============ Token name, symbol & URI ============ */
+
+    /**
+     * @dev Gets the token name.
+     * @return string representing the token name
+     */
+    function name() public pure override(ERC721Upgradeable, IKintoAppRegistry) returns (string memory) {
+        return "Kinto APP";
+    }
+
+    /**
+     * @dev Gets the token symbol.
+     * @return string representing the token symbol
+     */
+    function symbol() public pure override(ERC721Upgradeable, IKintoAppRegistry) returns (string memory) {
+        return "KINTOAPP";
+    }
+
+    /**
+     * @dev Returns the base token URI. ID is appended
+     * @return token URI.
+     */
+    function _baseURI() internal pure override returns (string memory) {
+        return "https://kinto.xyz/metadata/kintoapp/";
+    }
+
+    /* ============ App Registration ============ */
+
+    /**
+     * @dev Register a new app and mints the NFT to the creator
+     * @param _name The name of the app
+     * @param parentContract The address of the parent contract
+     * @param childContracts The addresses of the child contracts
+     * @param appLimits The limits of the app
+     */
+    function registerApp(
+        string calldata _name,
+        address parentContract,
+        address[] calldata childContracts,
+        uint256[4] calldata appLimits
+    ) external override {
+        require(appLimits.length == 4, "Invalid app limits");
```

Comment:
is this check redundant? if `appLimits` is already a fixed size array, wouldn't `.length` return always 4?

Reply:
yes, good catch